h2o-genmodel-extensions/jgrapht/src/main/java/hex/genmodel/tools/JgraphtPrintMojo.java

```diff
@@ -0,0 +1,261 @@
+        validateArgs();
+     
+        if (genModel instanceof TreeBackedMojoModel){
+            TreeBackedMojoModel treeBackedModel = (TreeBackedMojoModel) genModel;
+            ConvertTreeOptions options = new ConvertTreeOptions().withTreeConsistencyCheckEnabled();
+            final SharedTreeGraph g = treeBackedModel.convert(treeToPrint, null, options);
+            printPng(g);
+        }
+        else {
+            System.out.println("ERROR: Unknown MOJO type");
+            System.exit(1);
+        }
+    }
+
+    private void printPng(SharedTreeGraph trees) throws IOException, ImportException {
+        for (SharedTreeSubgraph tree : trees.subgraphArray) {
+            Path dotSourceFilePath = Files.createTempFile("", tmpOutputFileName);
+            try (FileOutputStream fosTemp = new FileOutputStream(dotSourceFilePath.toFile()); PrintStream osTemp = new PrintStream(fosTemp)) {
+                tree.printDot(osTemp, maxLevelsToPrintPerEdge, detail, optionalTitle, pTreeOptions, true);
+                generateOutputPng(dotSourceFilePath, (trees.subgraphArray.size() == 1 ? "" : tree.name));
+            }
+            Files.delete(dotSourceFilePath);
+        }
+
+    }
+
+    private void generateOutputPng(Path dotSourceFilePath, String treeName) throws ImportException, IOException {
+        LabeledVertexProvider vertexProvider = new LabeledVertexProvider();
+        LabeledEdgesProvider edgesProvider = new LabeledEdgesProvider();
+        ComponentUpdater componentUpdater = new ComponentUpdater();
+        DOTImporter<LabeledVertex, LabeledEdge> importer = new DOTImporter<>(vertexProvider, edgesProvider, componentUpdater);
+        DirectedMultigraph<LabeledVertex, LabeledEdge> result = new DirectedMultigraph<>(LabeledEdge.class);
+        try (FileInputStream is = new FileInputStream(dotSourceFilePath.toFile())) {
+            importer.importGraph(result, new InputStreamReader(is));
+            JGraphXAdapter<LabeledVertex, LabeledEdge> graphAdapter = new JGraphXAdapter<LabeledVertex, LabeledEdge>(result);
+            mxIGraphLayout treeLayout = new mxCompactTreeLayout(graphAdapter, true);
+            mxIGraphLayout nonOverlappingEdgesLayout = new mxEdgeLabelLayout(graphAdapter);
+            treeLayout.execute(graphAdapter.getDefaultParent());
+            nonOverlappingEdgesLayout.execute(graphAdapter.getDefaultParent());
+            BufferedImage image = mxCellRenderer.createBufferedImage(graphAdapter, null, 2, Color.WHITE, true, null);
+            if (outputFileName != null) {
+                ImageIO.write(image, "PNG", new File(getPngName(treeName)));
+            } else {
+                ImageIO.write(image, "PNG", System.out);
+            }
+        }
+    }
+
+    protected String getPngName(String treeName) {
+        if (treeName.isEmpty()) {
+            return outputFileName;
+        } else {
+            String infixTreeName = treeName.replaceAll("\\s+", "").replaceAll(",", "_");
```

Comment:
this magic is unnecessary imho, I think in case we want to generate multiple files the outputFile parameter should be treated as directory (created if not existing) and we should create treeX.png files there

Reply:
will fix

Reply:
Saving to directory done, left .replaceAll("\\s+", "").replaceAll(",", "_") part because otherwise names like "Tree 1, Class NO.png" would be possible